src/http/web_socket.cr

```diff
@@ -127,6 +182,20 @@ class HTTP::WebSocket
     size
   end
 
+  private def read_payload(buffer)
+    count = Math.min(@remaining, buffer.size)
+    if masked?
+      count.times do |i|
+        buffer[i] = @io.read_byte.not_nil! ^ @mask[@mask_offset % 4]
+        @mask_offset += 1
+      end
```

Comment:
Reading byte per byte is pretty slow. It's much faster to use `read_fully` and then unmask, like it was done before your changes. Could you revert it to that?